packages/core/src/actions/contracts/readContracts.ts

```diff
@@ -37,26 +37,36 @@ export async function readContracts<Data extends any[] = Result[]>({
 }: ReadContractsConfig): Promise<ReadContractsResult<Data>> {
   try {
     const provider = getProvider()
+
+    // We want to group the contracts by chain id so we can send it off
+    // in batch to the multicall contract.
+    // We also want to preserve the positions (index) of the `contracts` argument
+    // so we can reorder it later.
     const contractsByChainId = contracts.reduce<{
-      [chainId: number]: ReadContractsConfig['contracts']
-    }>((contracts, contract) => {
+      [chainId: number]: {
+        contract: ReadContractsContract
+        index: number
+      }[]
+    }>((contracts, contract, index) => {
```

Comment:
Assuming yes, but do we need this generic here or can it be inferred from `reduce` return type?